### PR TITLE
Catalogs actions

### PR DESCRIPTION
**Pull request by hamishs**
* Introduces a base class `CatalogObject` for catalog entries that includes a method to load them from an xml element. This can be used for general xml elements in the catalog as well as catalog entries e.g. bounding boxes. With this we can define custom entries by subclassing `CatalogEntry`.
* Adds subclasses of `Entity` for different catalog types (Vehicle, Pedestrian). In this way we can define custom entities like traffic lights. A subclass of `Entity` requires a type hint for its catalog entry argument which will indicate which `CatalogEntry` subclass should be used.
* When reading a catalog we can supply a list of `entity_types` which will be used to load each catalog entry. With this we can support loading custom entities from OpenSCENARIO.  
* Actions are defined by the `ScenarioAction` class. This has an abstract method `_apply` which can be implemented to apply the action at the given time to the state or the entity. A simple `UpdateStateVariableAction` action is provided which will iterate through the action variables and apply them to the entity.

### CODE REVIEW

File: scenario_gym/scenario/actions.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        t : float
+            The time at which the action should be applied.
+
+        action_class : str
+            The name of the action class.
+
+        entity_ref : str
+            Reference of the entity to which the action applies.
+
+        action_variables : str
```
Comment (nilsgoldbeck) [inline scenario_gym/scenario/actions.py:38]: I think the type should be `Dict[str, Any]` here

File: scenario_gym/scenario/actions.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+class UpdateStateVariableAction(ScenarioAction):
+    """Action that sets state variables for the entity."""
+
+    def _apply(
+        self, state: "State", entity: Optional[Entity]  # noqa: F821
+    ) -> None:
+        """Update the entity with action variables."""
+        for k, v in self.action_variables.items():
+            setattr(entity, k, v)
```
[the remark this answers is not in the document]
Comment (hamishs) [reply scenario_gym/scenario/actions.py:79]: Should we raise errors in either case? `entity` can be `None` when there is no entity in the scenario with the reference supplied by the action.

[the remark this answers is not in the document]
Comment (nilsgoldbeck) [reply scenario_gym/scenario/actions.py:79]: Not sure whether it's best to raise errors or silently ignore. How are issue with invalid scenario defintions handled elsewhere? Is there already a system for recording debug messages? If so, we could raise a warning for this.

[the remark this answers is not in the document]
Comment (hamishs) [reply scenario_gym/scenario/actions.py:79]: I have been raising warnings so that is probably a good convention to keep

**Pull request merged**